misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/command/BindingsGenerator.java

```diff
@@ -56,90 +58,121 @@
+                newSet.removeAll(existingBindings);
                 allJavaClasses.addAll(newSet);
                 classListForLooping.clear();
                 generateBindings(newSet, classLoader, dependenciesPath);
             }
-            createDirectory(utilsDirPath);
-            writeOutputFile(null, DEFAULT_TEMPLATE_DIR, JOBJECT_TEMPLATE_NAME,
-                    Paths.get(utilsDirPath, JOBJECT_FILE_NAME).toString(), false);
-            writeOutputFile(null, DEFAULT_TEMPLATE_DIR, ARRAY_UTILS_TEMPLATE_NAME,
-                    Paths.get(utilsDirPath, ARRAY_UTILS_FILE_NAME).toString(), false);
-
-            Path constantsPath = Paths.get(utilsDirPath, CONSTANTS_FILE_NAME);
-            Set<String> names = new HashSet<>(allClasses);
-            if (constantsPath.toFile().exists()) {
-                getUpdatedConstantsList(constantsPath, names);
-                notifyExistingDependencies(classNames, dependenciesPath.toFile());
-            }
-            writeOutputFile(names, DEFAULT_TEMPLATE_DIR, CONSTANTS_TEMPLATE_NAME, constantsPath.toString(), true);
 
+            // Generate the required util files.
+            generateUtilFiles();
+
+            // Handle failed binding generations.
             if (failedClassGens != null) {
-                errStream.print("\n");
-                for (String className : failedClassGens) {
-                    if (classNames.contains(className)) {
-                        errStream.println("Bindings for '" + className + "' class could not be generated.");
-                    }
-                    String simpleClassName = className.substring(className.lastIndexOf('.') + 1);
-                    writeOutputFile(className, DEFAULT_TEMPLATE_DIR, EMPTY_OBJECT_TEMPLATE_NAME,
-                            Paths.get(modulePathString, DEPENDENCIES_DIR_NAME,
-                                    simpleClassName + BAL_EXTENSION).toString(), false);
-                }
+                handleFailedClassGens();
             }
-            if (classLoader instanceof URLClassLoader) {
-                try {
-                    ((URLClassLoader) classLoader).close();
-                } catch (IOException ignored) {
-                }
+        }
+    }
+
+    private ClassLoader setClassLoader() throws BindgenException {
+
+        ClassLoader classLoader;
+        try {
+            if (!this.classPaths.isEmpty()) {
+                classLoader = getClassLoader(this.classPaths, this.getClass().getClassLoader());
+            } else {
+                outStream.println("No classpaths were detected.");
+                classLoader = this.getClass().getClassLoader();
+            }
+        } catch (BindgenException e) {
+            throw new BindgenException("Error while loading the classpaths.", e);
+        }
+        return classLoader;
+    }
+
+    private void setDirectoryPaths() {
+
+        if (outputPath == null) {
```

Comment:
If you use temp variable to hold the outputPath var and set its values to userDir if null, then we get remove duplicate codes.

Reply:
Simplified the code with a temp variable.